src/input.rs

```diff
@@ -569,6 +566,24 @@ mod tests {
         assert_eq!(err.to_string().contains(filename), true);
     }
 
+    #[test]
+    fn test_input_gzip() {
+        use std::io::Cursor;
+        let data = Cursor::new(
+            // echo '' | gzip -cf | od -A n -t u1 -w40| sed -e 's/^  *//; s/  */, /g'
+            vec![31, 139, 8, 0, 0, 0, 0, 0, 0, 3, 227, 2, 0, 147, 6, 215, 50, 1, 0, 0, 0],
+        );
+        let mut input = Input::open_stream(&PathBuf::from("test.log.gz"), Box::new(data)).unwrap();
+        let mut buf = [0; 32];
+        let result = input.stream.read(&mut buf);
```

Comment:
There are no guarantees how many bytes `read` will read except that it is `>=1` if there at least one byte in the input remains. That's why I would recommend to use `read_to_end`.
Also, it would be useful to check the content of the resulting buffer.